clang/test/CheckedCRewriter/basic_return_itype.c

```diff
@@ -55,7 +55,7 @@ int main() {
   bp1 = funcdecl(ap1, bp1, cp1);
   return 0;
 }
-//CHECK: int main() {
+//CHECK: int main(void) {
```

Comment:
This is interesting. In some sense we really want `main()` and not `main(void)` since `main` actually takes multiple arguments. It's just that `main` can't be declared `_Checked`. Can you remind me why we are adding ` void` again? Is it to enable adding `_Checked`

Reply:
Yes that's one of the reasons (issue 183). 
Aside from adding checked regions, if we try to return a struct with a checked type out of a function without a prototype, the compiler yells at us again (issue 182). Issue 182 is a pretty specific case, so for all intents and purposes, it is for Checked regions.
What's interesting here (that I just realized) is that you can't return a checked pointer type out of a function like `main` ever anyway, and the caveat from issue 183 is that "a function with no prototype cannot have a return type **that is a checked type**", i.e., having `int main() _Checked {` is fine! So I think we could make this PR better if I also added a check to ensure that the return type of the function is a pointer. Then we can cut down on some of the funkiness.

Reply:
Yes, that would be ideal. So perhaps go ahead and update the fix and send up another PR?